cabal-install/src/Distribution/Client/ProjectConfig/Lens.hs

```diff
@@ -0,0 +1,51 @@
+module Distribution.Client.ProjectConfig.Lens where
+
+import Distribution.Compat.Prelude
+import Distribution.Client.ProjectConfig.Types (ProjectConfig (..), ProjectConfigBuildOnly, ProjectConfigShared, ProjectConfigProvenance, PackageConfig, MapMappend)
+import Distribution.Compat.Lens
+import Distribution.Client.Types.SourceRepo (SourceRepoList)
+import qualified Distribution.Client.ProjectConfig.Types as T
+import Distribution.Package
+         ( PackageName, PackageId, UnitId )
+import Distribution.Types.PackageVersionConstraint
+         ( PackageVersionConstraint )
+
+projectPackages :: Lens' ProjectConfig [String]
+projectPackages f s = fmap (\x -> s { T.projectPackages = x }) (f (T.projectPackages s))
+{-# INLINE projectPackages #-}
```

Comment:
You may want to double-check that these bindings get into the interface file ([as shown here, for example](https://markkarpov.com/tutorial/ghc-optimization-and-fusion.html#how-to-control-inlining)). I think that the way to guarantee it is to use INLINABLE rather than INLINE. But maybe I’m wrong.

Reply:
As I've never worked with Lens before I just adapted the contents from module Distribution.Types.GenericPackageDescription.Lens here. But I will read into it, thank you!